Check `java.lang.ClassCastException` on socket tests

> Testing with #245 and #247 the tests on sockets reports:
> 
> ```
> java.lang.IllegalStateException: Failed to cast class java.lang.Integer into int
> 	at system.net.sockets.Socket.getAvailable(Socket.java:1975)
> 	at nettest.HelloNETSocketClient.StartClient(HelloNETSocketClient.java:97)
> 	at nettest.HelloNETSocket$2.Invoke(HelloNETSocket.java:64)
> 	at system.threading.ThreadStart.DelegateInvoked(ThreadStart.java:137)
> Caused by: java.lang.ClassCastException
> ```
> 
> Another enhancement can be something like:
> 1. if `java.lang.ClassCastException` is raised...
> 2. ... try to cast the value to `java.lang.Number`...
> 3. ...then use the function to return the native type
> 4. if point 2 fails emit the first exception: e.g. **java.lang.IllegalStateException: Failed to cast class java.lang.Integer into int** 

 _Originally posted by @masesdevelopers in [#246](https://github.com/masesgroup/JCOReflector/issues/246#issuecomment-3281199723)_

With changes on code from #247 and #249 the following callstack is raised:

```
java.lang.IllegalStateException: Failed to convert class java.lang.Integer into int and, as fallback solution, into java.lang.Number
	at system.net.sockets.Socket.getAvailable(Socket.java:2451)
	at nettest.HelloNETSocketClient.StartClient(HelloNETSocketClient.java:97)
	at nettest.HelloNETSocket$2.Invoke(HelloNETSocket.java:64)
	at system.threading.ThreadStart.DelegateInvoked(ThreadStart.java:137)
Caused by: java.lang.ClassCastException
```

The issue is not raised neither on start nor on end, but randomly within execution of socket test in async mode and always for the same function.

--- a/src/net/engine/JCObjectReflectorClassNativeGetPropertyWithCastToNumber.template
+++ b/src/net/engine/JCObjectReflectorClassNativeGetPropertyWithCastToNumber.template
@@ -6,6 +6,7 @@
             retObjectPROPERTY_NAME = PROPERTY_OBJECT.Get("PROPERTY_NAME"PROPERTY_INVOKE_PARAMETERS);
             return (PROPERTY_OUTPUTTYPE)retObjectPROPERTY_NAME;
         } catch (java.lang.ClassCastException cce) {
+            boolean reportPROPERTY_NAMEError = true;
             java.lang.String retObjectPROPERTY_NAME_ToString = retObjectPROPERTY_NAME == null ? "null" : retObjectPROPERTY_NAME.toString();
             try {
                 if (retObjectPROPERTY_NAME != null) {
@@ -21,7 +22,13 @@
                 }
                 else throw new java.lang.NullPointerException("Return value is null and this is not expected");
             } catch (java.lang.Exception cceInner) {
+                reportPROPERTY_NAMEError = false;
                 throw new java.lang.IllegalStateException(java.lang.String.format("Failed to convert %s (%s) into PROPERTY_OUTPUTTYPE and, as fallback solution, using java.lang.Number with exception %s (%s)", retObjectPROPERTY_NAME != null ? retObjectPROPERTY_NAME.getClass() : "null", retObjectPROPERTY_NAME_ToString, cceInner.getClass(), cceInner.getMessage()), cce);
+            }
+            finally {
+                if (reportPROPERTY_NAMEError) {
+                    java.lang.System.err.println("Output returned from a fallback solution.");
+                }
             }
         } catch (JCNativeException jcne) {
             throw translateException(jcne);

--- a/src/net/engine/JCObjectReflectorClassNativeMethodWithCastToNumber.template
+++ b/src/net/engine/JCObjectReflectorClassNativeMethodWithCastToNumber.template
@@ -6,6 +6,7 @@
             retObjectMETHOD_NAME = METHOD_OBJECT.Invoke("METHOD_NAME"METHOD_INVOKE_PARAMETERS);
             return (METHOD_RETURN_TYPE)retObjectMETHOD_NAME;
         } catch (java.lang.ClassCastException cce) {
+            boolean reportMETHOD_NAMEError = true;
             java.lang.String retObjectMETHOD_NAME_ToString = retObjectMETHOD_NAME == null ? "null" : retObjectMETHOD_NAME.toString();
             try {
                 if (retObjectMETHOD_NAME != null) {
@@ -21,7 +22,13 @@
                 }
                 else throw new java.lang.NullPointerException("Return value is null and this is not expected");
             } catch (java.lang.Exception cceInner) {
+                reportMETHOD_NAMEError = false;
                 throw new java.lang.IllegalStateException(java.lang.String.format("Failed to convert %s (%s) into METHOD_RETURN_TYPE and, as fallback solution, using java.lang.Number with exception %s (%s)", retObjectMETHOD_NAME != null ? retObjectMETHOD_NAME.getClass() : "null", retObjectMETHOD_NAME_ToString, cceInner.getClass(), cceInner.getMessage()), cce);
+            }
+            finally {
+                if (reportMETHOD_NAMEError) {
+                    java.lang.System.err.println("Output returned from a fallback solution.");
+                }
             }
         } catch (JCNativeException jcne) {
             throw translateException(jcne);

